torch/distributed/fsdp/_flat_param.py

```diff
@@ -862,8 +862,9 @@ def shard(self):
             start_idx = sharded_flat_param.numel() * self.rank
             end_idx = sharded_flat_param.numel() * (self.rank + 1) - 1  # inclusive
             self._init_shard_metadata(numel_padded, start_idx, end_idx)
-            if orig_storage._size() > 0:
-                orig_storage._resize_(0)
+            if not torch.distributed._functional_collectives.is_torchdynamo_compiling():
```

Comment:
a comment here would also be helpful, perhaps cross referencing a `Note [Skip storage resize when Dynamo tracing in FSDP]`